src/lib/utils/fetchGitHubContributors.ts

```diff
@@ -0,0 +1,42 @@
+import { join } from "path"
+import { GITHUB_AUTH_HEADERS, GITHUB_COMMITS_URL } from "@/lib/constants"
+import type { Author } from "@/lib/interfaces"
+import type { FileContributorsState } from "@/lib/types"
+
+export const fetchGitHubAuthors = async (
+  relativePath: string
+): Promise<FileContributorsState> => {
+  const url = new URL(GITHUB_COMMITS_URL)
+  const CONTENT_DIR = "/src/content/" // TODO: Overriding with Gatsby repo—Remove and import from constants.ts for production
+  const filePath = join(CONTENT_DIR, relativePath, "index.md")
+  url.searchParams.set("path", filePath)
+
+  try {
+    const uniqueAuthors: Array<Author> = []
+    const response = await fetch(url, GITHUB_AUTH_HEADERS)
+    const commits = await response.json()
+    const authorSet = new Set<string>()
+    commits
+      .filter(({ author }) => author)
+      .forEach(({ author, commit }) => {
+        const entry: Author = {
+          name: commit.author.name,
+          email: commit.author.email,
+          avatarUrl: author.avatar_url,
+          user: {
+            login: author.login,
+            url: author.html_url,
+          },
+        }
+        // Unique authors only
+        authorSet.add(JSON.stringify(entry))
```

Comment:
Why do you need to stringify it?

Reply:
@pettinarip To compare by value and not by reference... I'm adding to a Set to make sure everything is unique... if I just add `entry`, it's added by memory reference and two matching objects are being seen as different entries.

Reply:
Ah right! thanks for explaining.